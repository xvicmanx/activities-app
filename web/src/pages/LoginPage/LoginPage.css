.LoginPage {
  padding: 10px;
  background: #919ba2;
  min-height: 100vh;
}

.LoginPage__Container {
  background: #fff;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
  overflow: hidden;
}

.LoginPage__Content{
  width: 100%;
  background: #f9f9f9;  
}

.LoginPage__App-Info {
  text-align: center;
  background: #fff;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  border-bottom: 1px solid #efefef;
  border-right: 1px solid #efefef;
}

.LoginPage__Form {
  width: 100%;
  height: calc(100% - 55px);
  padding: 0 0 20px 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.LoginPage__Logo.top {
  display: none;
}

.LoginPage__Title,
.LoginPage__Subtitle {
  color: #919ba2;
}

.LoginPage__Subtitle {
  font-size: 2rem !important;
  font-weight: 900;
  line-height: 1.6;
}

@media only screen and (min-width: 1024px) {
  .LoginPage {
    padding: 40px;
  }

  .LoginPage__Logo.top {
    margin: auto;
    width: 50%;
    display: block;
  }

  .LoginPage__Form {
    padding: 20px 80px;
  }

  .LoginPage__Container {
    display: flex;
    height: calc(100vh - 80px);
  }

  .LoginPage__App-Info {
    padding: 70px 60px;
    max-width: 500px;
  }

  .LoginPage__Subtitle {
    font-size: 1.25rem;
  }

}