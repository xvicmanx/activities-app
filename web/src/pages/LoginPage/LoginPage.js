import React from 'react';
import { Heading, Image } from 'react-bulma-components';

import './LoginPage.css';

import { Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { LoginForm } from '../../forms';
import { NavBar } from '../../components';
import logo from "../../assets/images/logo.png";

const LoginPage = () => {
  const { Users } = useSelector(state => state);

  if (Users.data) {
    return <Redirect to="/home" />;
  }

  return (
    <div className="LoginPage">
      <div className="LoginPage__Container">
        <div className="LoginPage__App-Info">
          <div className="LoginPage__LogoContainer top">
            <Image
              src={logo}
              alt="Pic"
              className="LoginPage__Logo top"
            />
          </div>
          <Heading subtitle className="LoginPage__Subtitle">
            Insiemi
          </Heading>
        </div>
        
        <div className="LoginPage__Content">
          <NavBar right />
          <div className="LoginPage__Form">
            <LoginForm />
          </div>
        </div>
      </div>
    </div>
  );
};

export default LoginPage;
