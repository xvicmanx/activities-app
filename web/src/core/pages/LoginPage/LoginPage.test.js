import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

jest.mock('../../forms', () => ({
  LoginForm: () => 'LoginForm',
}));

jest.mock('react-router-dom', () => ({
  Redirect: () => 'Redirect',
}));

import LoginPage from './LoginPage';


Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore();

describe('LoginPage', () => {
  it('renders properly when user is not signed in', () => {
    const store = mockStore({
      Users: { data: null },
    });
    const result = Enzyme.mount(
      <Provider store={store}>
        <LoginPage />
      </Provider>
    );
    expect(result.find('.LoginPage').props()).toMatchSnapshot();
  });

  it('renders properly when user is signed in', () => {
    const store = mockStore({
      Users: { data: {} },
    });
    const result = Enzyme.mount(
      <Provider store={store}>
        <LoginPage />
      </Provider>
    );
    expect(result.find('Redirect').props()).toMatchSnapshot();
  });
});
