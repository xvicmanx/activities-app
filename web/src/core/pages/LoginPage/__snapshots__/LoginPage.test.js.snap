// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders properly when user is not signed in 1`] = `
Object {
  "children": <div
    className="LoginPage__Container"
  >
    <div
      className="LoginPage__App-Info"
    >
      <div
        className="LoginPage__LogoContainer top"
      >
        <Image
          alt="Pic"
          className="LoginPage__Logo top"
          fallback="https://bulma.io/images/placeholders/480x480.png"
          rounded={false}
          src="logo.png"
        />
      </div>
      <Heading
        className="LoginPage__Subtitle"
        heading={false}
        renderAs="h1"
        spaced={false}
        subtitle={true}
      >
        Insiemi
      </Heading>
    </div>
    <div
      className="LoginPage__Content"
    >
      <div
        className="LoginPage__Form"
      >
        <LoginForm />
      </div>
    </div>
  </div>,
  "className": "LoginPage",
}
`;

exports[`LoginPage renders properly when user is signed in 1`] = `
Object {
  "to": "/home",
}
`;
